File: src/python/factorial_recursive.py
"""Recursive factorial algorithm"""


def factorial_recursive(num):
    """
    Implementation of factorial algorithm using recursion

    Arguments:
        num: int. the number of which you want to obtain the factorial.

    Returns the result of the factorial operation
    """
    if num == 1:
        return 1

    return num * factorial_recursive(num - 1)


print(factorial_recursive(5))

File: src/java/FactorialRecursive.java
public class FactorialRecursive {

  public static void main(String[] args) {
    System.out.println("Fatorial de 0 : " + FactorialRecursive(0));
    System.out.println("Fatorial de 1 : " + FactorialRecursive(1));
    System.out.println("Fatorial de 3 : " + FactorialRecursive(3));
    System.out.println("Fatorial de 7 : " + FactorialRecursive(7));
  }

  public static int FactorialRecursive(int x) {
    if (x <= 1) return 1;
    else return x * FactorialRecursive(x - 1);
  }
}
